### PR TITLE
[cooperative perception] Add covariance calculations to SDSM conversions

### PR DESCRIPTION
**Pull request by adamlm**
# PR Details
## Description

This SDSM to detected object list conversion did not generate the covariance matrix values from confidence intervals, making the resulting detection list pretty much useless.

## Related Jira Key

Closes [CDAR-729](https://usdot-carma.atlassian.net/browse/CDAR-729)

## Motivation and Context

SDSM to detection list conversion is necessary to receive detected object information from nearby actors.

## How Has This Been Tested?

Manually in integration testing.

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.

[CDAR-729]: https://usdot-carma.atlassian.net/browse/CDAR-729?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: carma_cooperative_perception/include/carma_cooperative_perception/sdsm_to_detection_list_component.hpp [hunk 1]
```diff
@@ -47,7 +47,11 @@ class SdsmToDetectionListNode : public carma_ros2_utils::CarmaLifecycleNode
 
   auto sdsm_msg_callback(const input_msg_type & msg) const -> void
   {
-    publisher_->publish(to_detection_list_msg(msg, georeference_));
+    try {
+      publisher_->publish(to_detection_list_msg(msg, georeference_));
+    } catch (const std::runtime_error & e) {
+      RCLCPP_ERROR_STREAM(get_logger(), "Not converting SDSM to detection list: " << e.what());
```
Comment (paulbourelly999) [inline carma_cooperative_perception/include/carma_cooperative_perception/sdsm_to_detection_list_component.hpp:53]: Low priority. Seems like bad logging language. Maybe failed to convert SDSM to detection list

File: carma_cooperative_perception/src/msg_conversion.cpp [hunk 2]
```diff
@@ -267,6 +269,23 @@ auto to_detection_list_msg(
       ref_pos_map.easting + pos_offset.offset_x, ref_pos_map.northing + pos_offset.offset_y,
       ref_pos_map.elevation + pos_offset.offset_z.value_or(units::length::meter_t{0.0})});
 
+    // Pose covariance is x, y, z, roll, pitch, yaw
+    try {
+      detection.pose.covariance.at(0) =
```
Comment (paulbourelly999) [inline carma_cooperative_perception/src/msg_conversion.cpp:274]: Is covariance a flattened out 3X3 matrix?

File: carma_cooperative_perception/src/msg_conversion.cpp [hunk 3]
```diff
@@ -281,6 +300,14 @@ auto to_detection_list_msg(
     quat_tf.setRPY(0, 0, remove_units(units::angle::radian_t{enu_yaw}));
     detection.pose.pose.orientation = tf2::toMsg(quat_tf);
 
+    try {
+      // Pose covariance is x, y, z, roll, pitch, yaw
```
Comment (paulbourelly999) [inline carma_cooperative_perception/src/msg_conversion.cpp:304]: Might be helpful to explicitly mention at these lines that we are dealing with flattened 6x6 matrix

**Review by paulbourelly999 — DISMISSED** [review]

File: carma_cooperative_perception/include/carma_cooperative_perception/sdsm_to_detection_list_component.hpp (hunk 1 above)
Comment (adamlm) [reply carma_cooperative_perception/include/carma_cooperative_perception/sdsm_to_detection_list_component.hpp]: Updated

File: carma_cooperative_perception/src/msg_conversion.cpp (hunk 2 above)
Comment (adamlm) [reply carma_cooperative_perception/src/msg_conversion.cpp:274]: Flattened out 6x6 matrix. I updated the comment to be more descriptive.

File: carma_cooperative_perception/src/msg_conversion.cpp (hunk 3 above)
Comment (adamlm) [reply carma_cooperative_perception/src/msg_conversion.cpp]: Updated

**Review by paulbourelly999 — APPROVED** [review]

**Pull request merged**